pkg/core/native/designate.go

```diff
@@ -384,10 +412,6 @@ func (s *Designate) DesignateAsRole(ic *interop.Context, r noderoles.Role, pubs
 		return fmt.Errorf("failed to update Designation role data cache: %w", err)
 	}
 
-	ic.AddNotification(s.Hash, DesignationEventName, stackitem.NewArray([]stackitem.Item{
```

Comment:
But why? I'd expect to see this notification.

Reply:
Because it's not a standard invocation. We put role values into storage not via transaction, but via standard process of contract initialisation, thus, role values should be treated as "defaults" for this network. Although notification would be useful here for some third-party network users.

IMO, notification on the roles initialisation doesn't suit well to the existing model, but we still can add it, nothing prevents us from it. Then the notification will be emitted during Genesis OnPersist, during native Management deployment, is it OK?

Reply:
Just like we have an event for GAS/NEO emission, mainnet 0:
```
      "blockhash" : "0x1f4d1defa46faa5e7b9b8d3f79a06bec777d7c26c4aa5f6f5899a291daa87c15",
      "executions" : [
         {
            "notifications" : [
               {
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "type" : "Any"
                        },
                        {
                           "value" : "axI92L7HGGSIUrvHhZXjU2oFj58=",
                           "type" : "ByteString"
                        },
                        {
                           "value" : "100000000",
                           "type" : "Integer"
                        }
                     ]
                  },
                  "eventname" : "Transfer",
                  "contract" : "0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5"
               },
               {
                  "eventname" : "Transfer",
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "type" : "Any"
                        },
                        {
                           "value" : "axI92L7HGGSIUrvHhZXjU2oFj58=",
                           "type" : "ByteString"
                        },
                        {
                           "value" : "5200000000000000",
                           "type" : "Integer"
                        }
                     ]
                  },
                  "contract" : "0xd2a4cff31913016155e38e474a2c06d08be276cf"
               }
            ],
            "trigger" : "OnPersist",
            "gasconsumed" : "0",
            "exception" : null,
            "stack" : [],
            "vmstate" : "HALT"
         },
```
We'll have some more with this extension. OK, what's bad about it?

Reply:
Fixed.